opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/tasks/instancedata/Overridden.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.provider.tasks.processors.tasks.instancedata;
+
+import android.content.ContentValues;
+
+import org.dmfs.iterables.elementary.Seq;
+import org.dmfs.jems.optional.decorators.Mapped;
+import org.dmfs.jems.single.Single;
+import org.dmfs.optional.NullSafe;
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.adapters.FirstPresent;
+import org.dmfs.rfc5545.DateTime;
+import org.dmfs.tasks.contract.TaskContract;
+
+
+/**
+ * A decorator for {@link Single}s of Instance {@link ContentValues} which populates the {@link TaskContract.Instances#INSTANCE_ORIGINAL_TIME} field based on
+ * the given {@link Optional} original start and the already populated {@link TaskContract.Instances#INSTANCE_START} and {@link
+ * TaskContract.Instances#INSTANCE_DUE} fields.
+ *
+ * @author Marten Gajda
+ */
+public final class Overridden implements Single<ContentValues>
```

Comment:
Just that I understand it correctly, this class maybe used when creating instances for an original task - in which case START or DUE is used, or creating more instances of a recurring task - in which case ORIGINAL_TIME is copied. Right?

Reply:
I that `isRecurring()` is checked before using this...
I don't have a clear picture of how recurring tasks will be originated. For example if we changed an existing non-recurring task to recurring, wouldn't this code result in `0` ORIGINAL_TIME for the recurring instance rows?

Reply:
`ORIGINAL_TIME` always refers to the "start" (or "due" in the absence of start) of the original instance as determined by the recurrence rule. For simplicity it can always be `0` for non-recurring tasks. If a task is converted to recurring the `0` value will be overridden with proper timestamps.
The `ORIGINAL_TIME` identifies the instance that's overridden by an override (there is no other "key" you can use for that, because the override might change the actual time).
So `ORIGINAL_TIME` is different for each instance of a recurring task. It's not the start of the first instance of a recurring task.